kythe/go/extractors/config/smoke/smoke.go

```diff
@@ -38,8 +38,7 @@ import (
 // automatically. The current implementation is to simply check the fraction of
 // files covered by extractor output.
 //
-// TODO(danielmoy): hook up an indexing step and check actual semantic output
-// instead of simple files.
+// Also optionally supports indexing the extractors output.
 //
 // TODO(danielmoy): support more than just java.
 type Tester interface {
```

Comment:
I'm not really convinced this needs to be an interface—it seems like it could be a plain function

    type Tester func(ctx context.Context, repo config.Repo) (Result, error)

Similarly with `Indexer` and `Fetcher` below. At some point it might make sense to make these configuration values on the smoke test constructor, see below. That makes it a little nicer to provide simple default implementations, e.g.,

```
func GitFetch(ctx context.Context, repo config.Repo) error {
  return exec.CommandContext(ctx, "git", "clone", repo.URI, repo.OutputPath).Run()
}
```

See also my comment at the constructor.